projects/auth0-angular/src/lib/auth.guard.ts

```diff
@@ -3,20 +3,41 @@ import {
   ActivatedRouteSnapshot,
   RouterStateSnapshot,
   CanActivate,
+  CanLoad,
+  Route,
+  UrlSegment,
+  CanActivateChild,
 } from '@angular/router';
 import { Observable, of } from 'rxjs';
-import { tap } from 'rxjs/operators';
+import { tap, take } from 'rxjs/operators';
 import { AuthService } from './auth.service';
 
 @Injectable({
   providedIn: 'root',
 })
-export class AuthGuard implements CanActivate {
+export class AuthGuard implements CanActivate, CanLoad, CanActivateChild {
   constructor(private auth: AuthService) {}
 
+  canLoad(route: Route, segments: UrlSegment[]): Observable<boolean> {
+    return this.auth.isAuthenticated$.pipe(take(1));
```

Comment:
The observable returned here must complete, hence the `take(1)`. `isAuthenticated$` does not complete by itself as it could revert back to an unauthenticated state later.

https://github.com/angular/angular/issues/18991#issuecomment-327151035

^ This is actually the same for `canActivate` too but those already return completed observables.

Reply:
I assume this observable is evaluated once per component attempt to load? otherwise, it would be stuck always evaluating to one state. We can discuss my doubts on our call tomorrow.

Reply:
Yep, just once. Piping creates a _new_ observable from `isAuthenticated$`, which is then completed using `take`. The original observable is untouched.